mycore-base/src/main/java/org/mycore/access/facts/condition/combined/MCRAbstractCombinedCondition.java

```diff
@@ -64,4 +68,14 @@ public void debugInfoForMatchingChildElement(MCRCondition c, boolean matches) {
         }
     }
 
+    boolean addDebugInfoIfRequested(MCRCondition c, MCRFactsHolder facts) {
+        if (!isDebug()) {
+            return c.matches(facts);
+        }
+        boolean matches = c.matches(facts);
```

Comment:
Maybe:
```
boolean matches = c.matches(facts);
if (!isDebug()) {
    return matches;
}
```
This would make it more obvious at a glance that debug- and non-debug-mode actually perform the same thing.